lib/lz4.c

```diff
@@ -124,9 +125,11 @@
 /**************************************
 *  Memory routines
 **************************************/
+#if (HEAPMODE)
```

Comment:
If there is a need to totally remove any reference to `<stdlib.h>`, then it will have to be done differently.
A few functions depend on them (typically `LZ4_createXXXX`). No `malloc()` means no way to use them.